it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,362 @@
+                    }
+                    postgresqlRdbmsBuilder.addPostgresqlSchemas(postgresqlSchemaBuilder);
+                }
+                sourceConfigBuilder.setPostgresqlSourceConfig(
+                        PostgresqlSourceConfig.newBuilder()
+                                .setIncludeObjects(postgresqlRdbmsBuilder)
+                                .setPublication(((PostgresSource) source).getPublication())
+                                .setReplicationSlot(((PostgresSource) source).getReplicationSlot())
+                );
+
+            case ORACLE:
+                OracleRdbms.Builder oracleRdmsBuilder = OracleRdbms.newBuilder();
+                for (String schema : allowedTables.keySet()) {
+                    OracleSchema.Builder oracleSchemaBuilder = OracleSchema.newBuilder().setSchema(schema);
+                    for (String table : allowedTables.get(schema)) {
+                        oracleSchemaBuilder.addOracleTables(OracleTable.newBuilder().setTable(table));
+                    }
+                    oracleRdmsBuilder.addOracleSchemas(oracleSchemaBuilder);
+                }
+                sourceConfigBuilder
+                        .setOracleSourceConfig(OracleSourceConfig.newBuilder().setIncludeObjects(oracleRdmsBuilder));
+        }
+
+        return sourceConfigBuilder.build();
+    }
+
+    @Override
+    public ConnectionProfile createGCSDestinationConnectionProfile(
+            String connectionProfileId, String gcsBucketName, String gcsRootPath) {
+
+        LOG.info("Creating GCS Destination Connection Profile {} in project {}.", connectionProfileId, projectId);
+
+        try {
+            ConnectionProfile.Builder connectionProfileBuilder = ConnectionProfile.newBuilder()
+                    .setDisplayName(connectionProfileId)
+                    .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance())
+                    .setGcsProfile(GcsProfile.newBuilder().setBucket(gcsBucketName).setRootPath(gcsRootPath));
+
+            CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                    .setParent(LocationName.of(projectId, location).toString())
+                    .setConnectionProfile(connectionProfileBuilder)
+                    .setConnectionProfileId(connectionProfileId)
+                    .build();
+
+            ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+            createdConnectionProfileIds.add(connectionProfileId);
+            return reference;
+        } catch (ExecutionException | InterruptedException e) {
+            throw new DatastreamResourceManagerException("Failed to create JDBC source connection profile. ", e);
+        }
+    }
+
+    public DestinationConfig buildGCSDestinationConfig(
+            String connectionProfileId,
+            String path,
+            String destinationOutputFormat) {
```

Comment:
I would change `destinationOutputFormat` to be either an enum with values  AVRO_FILE_FORMAT and JSON_FILE_FORMAT, or make it a boolean `use_avro` or something similarly named since there are only 2 options. Much clearer than comparing string literals.

And as a side note, when you do compare strings like you do on line 204 with `destinationOutputFormat == AVRO_FILE_FORMAT`, I think the standard, at least in our repo, is to use `.equals()`. Technically someone could pass a value for `destinationOutputFormat` that would cause this to not equal true even if they are the same value. Here is an example of that: https://stackoverflow.com/questions/8676775/string-comparison-in-java-what-is-wrong-with